shared/common/ui/customScrollbar/index.tsx

```diff
@@ -31,45 +33,57 @@ export function CustomScrollbars({
   verticalBarClass,
   innerClass,
   headerPadding = 0,
+  scrollIgnoreLength = 0,
+  bottomScrollBarOffset = 8,
   reverse,
   hideVertical,
   hideHorizontal,
 }: PropsWithChildren<CustomScrollbarsProps>) {
   const ref = useRef<HTMLDivElement>(null);
 
   const scrollSizes = useRef<[number, number]>(defaultScrollSizes);
+  const [rawScrollOffset, setRawScrollOffset] = useState(0);
   const [scrollOffsets, setScrollOffsets] = useState<[number, number]>(() => [
-    0,
-    0,
+    0, 0,
   ]);
   const _scrollOffsets = useRef(scrollOffsets);
   const [dragging, setDragging] = useState(false);
 
+  const scrollBarTopOffset =
+    rawScrollOffset < scrollIgnoreLength
+      ? rawScrollOffset
+      : scrollIgnoreLength;
+
   const onScroll = useCallback(
     (el: HTMLElement) => {
+      const scrollTop =
+        Math.max(0, el.scrollTop - scrollIgnoreLength) /
+        (el.scrollHeight - scrollIgnoreLength - el.clientHeight);
+
       _scrollOffsets.current = [
-        ((reverse
-          ? el.scrollHeight + el.scrollTop - el.clientHeight
-          : el.scrollTop) /
-          (el.scrollHeight - el.clientHeight)) *
```

Comment:
We need to keep this reverse scroll logic, so `scrollTop` should include this bit `reverse ? el.scrollHeight + el.scrollTop - el.clientHeight : el.scrollTop` in place of `el.scrollTop` I think.